libs/utils/utils.js

```diff
@@ -764,8 +764,9 @@ async function decorateIcons(area, config) {
   const { base } = config;
```

Comment:
From a performance standpoint, I'd suggest we do a very small check if there are any federal icons to begin with and start preloading the federal script:

```js
if([...icons].some(icon => icon.href.includes('/federal/')) loadLink(`...federal`, { rel: 'preload', as: 'fetch', crossorigin: 'anonymous' });
```

Reply:
I actually think we just load It since we need it on any span.icon and don't have the href yet anyhow. 
...I made an update to preload federated.js instead of the non used icons.svg on util decorateIcon()

```
 loadStyle(`${base}/features/icons/icons.css`);
 loadLink(`${base}/utils/federated.js`, { rel: 'preload', as: 'fetch', crossorigin: 'anonymous' });
```
LMK if see any issue w/ this. Thanks

Reply:
Nope that's good 👍

Reply:
It seems to load the file twice now though 🤔 
<img width="764" alt="Screenshot 2024-10-02 at 11 07 23" src="https://github.com/user-attachments/assets/88ab3f9c-af25-4fd8-8e1c-dfcd382a3604">
<img width="1137" alt="Screenshot 2024-10-02 at 11 07 15" src="https://github.com/user-attachments/assets/fbf20e70-2ec6-471b-8648-857fa93e50d9">